src/ObjectList.php

```diff
@@ -357,4 +359,19 @@ public function sum(callable $valueProvider, $startValue = null)
             return $total + $valueProvider($item);
         }, $startValue);
     }
+
+    /**
+     * @param callable $filter
+     * @return mixed
+     */
+    public function get(callable $filter)
+    {
+        foreach ($this as $item) {
+            if ($filter($item)) {
+                return $item;
+            }
+        }
+
+        throw new RuntimeException("Could not find an item in " . get_class($this) . " using a find filter.");
```

Comment:
My use case is selecting a particular leg of a particular journey - there has to be one and only one, so I do something like `$journeyList->getThatJourney()->getLegs()->getThatLeg()`. Otherwise I'd have to do something like `$journeyList->filterThatJourney()->getLegs()->getMatchingLegs()` which is possible, but requires working with lists and `getLegs()` would be an operation on a list of Journeys. Doable, just less straightforward. But maybe it should work like jQuery in that respect.